### ISSUE TITLE
RODFT with smear

### ISSUE THREAD

**Issue by edoapra** [nwchemgit/nwchem#616]
**Describe the bug**
If the keyword smear is used with RODFT, occupation numbers in the Molecular Orbital analysis are non-sensical.
Energy values seem to make sense.

**Attach log files**
```
 Vector    1  Occ=0.000000D+00  E=-1.036655D+01
              MO Center= -3.9D-18, -2.3D-17,  7.4D-19, r^2= 2.8D-02
   Bfn.  Coefficient  Atom+Function         Bfn.  Coefficient  Atom+Function
  ----- ------------  ---------------      ----- ------------  ---------------
     1      0.995386  1 C  s

 Vector    2  Occ=0.000000D+00  E=-5.458552D-01
              MO Center= -1.6D-16, -2.8D-16,  9.4D-17, r^2= 8.3D-01
   Bfn.  Coefficient  Atom+Function         Bfn.  Coefficient  Atom+Function
  ----- ------------  ---------------      ----- ------------  ---------------
     6      0.592905  1 C  s                  2      0.480770  1 C  s
     1     -0.231387  1 C  s

 Vector    3  Occ=0.000000D+00  E=-2.001641D-01
              MO Center=  1.3D-18, -4.5D-20, -9.2D-18, r^2= 1.0D+00
   Bfn.  Coefficient  Atom+Function         Bfn.  Coefficient  Atom+Function
  ----- ------------  ---------------      ----- ------------  ---------------
     5      0.623577  1 C  pz                 9      0.492665  1 C  pz

 Vector    4  Occ=5.643769D-01  E=-1.207348D-01
              MO Center=  3.5D-17, -3.5D-19,  4.8D-18, r^2= 1.1D+00
```

**To Reproduce**
1. Steps to reproduce the behavior:
Run this C atom input file
```
geometry
  C   0.0 0.0 0.0
end

basis
  * library 6-31G
end

dft
  xc pbe0
  smear
  rodft
end
task dft energy
```

**Expected behavior**
Print occupation number in the range 0 to 2, with core orbital having a value of 2.

**Comment by edoapra** [comment]
See https://groups.google.com/g/nwchem-forum/c/O5atoUjoJK8

**Comment by edoapra** [comment]
On closer inspection, SMEAR is not correctly interfaces with CG/RODFT.
Commit https://github.com/nwchemgit/nwchem/issues/616 stops CG when the keyword `SMEAR` is used

**Comment by ice-hanbin** [comment]
I delete the keyword "smear" and keep the "rodft", the occupation number is still wrong
`
                       DFT Final Molecular Orbital Analysis
                       ------------------------------------

 Vector    1  Occ=5.643769D-01  E=-1.036655D+01
              MO Center=  4.1D-18, -7.4D-19,  1.8D-18, r^2= 2.8D-02
   Bfn.  Coefficient  Atom+Function         Bfn.  Coefficient  Atom+Function
  ----- ------------  ---------------      ----- ------------  ---------------
     1      0.995386  1 C  s

 Vector    2  Occ=************  E=-5.458552D-01
              MO Center=  1.4D-16, -1.9D-16,  3.0D-16, r^2= 8.3D-01
   Bfn.  Coefficient  Atom+Function         Bfn.  Coefficient  Atom+Function
  ----- ------------  ---------------      ----- ------------  ---------------
     6      0.592905  1 C  s                  2      0.480770  1 C  s
     1     -0.231387  1 C  s

 Vector    3  Occ=0.000000D+00  E=-2.001641D-01
              MO Center=  1.9D-17, -3.1D-17, -2.2D-16, r^2= 1.0D+00
   Bfn.  Coefficient  Atom+Function         Bfn.  Coefficient  Atom+Function
  ----- ------------  ---------------      ----- ------------  ---------------
     5      0.620711  1 C  pz                 9      0.490401  1 C  pz

`

**Comment by ice-hanbin** [comment]
I delete the keyword "smear" but keep the "rodft", the occupation number is still wrong
[c.log](https://github.com/nwchemgit/nwchem/files/8704694/c.log)

**Comment by ice-hanbin** [comment]
But I think this should not happen in principle, so how can I fix the bug?

**Comment by ice-hanbin** [comment]
But I don't think this should happen in principle, so how can I fix the bug?

**Comment by edoapra** [comment]
Your run is a `restart` run, therefore the `smear` keyword was kept from the first run.
Please add the `start` keyword at the beginning of the input file.
[https://nwchemgit.github.io/Start_Restart.html](https://nwchemgit.github.io/Start_Restart.html)

```
    input           = c.inp
    prefix          = c.
    data base       = ./c.db
    status          = restart
    nproc           =        4
    time left       =     -1s
```
Modified input file
```
start c_atom_rodft

geometry
  C   0.0 0.0 0.0
end

basis
  * library 6-31G
end

dft
  xc pbe0
 # smear
  rodft
end
task dft energy
```

**Comment by ice-hanbin** [comment]
Sorry about that, I have found this, but the problem is still here, SMEAR is not correctly interfaced with CG/RODFT, so I'd like to ask if there are some ways to fix this bug?

**Comment by edoapra** [comment]
Yes, the bug could be fixed. However, I am not sure any NWChem developer is going to act on it anytime soon.

**Comment by edoapra** [comment]
The `FON` [keyword](https://nwchemgit.github.io/Density-Functional-Theory-for-Molecules.html#fon-calculations-with-fractional-numbers-of-electrons) is also affected by the fact that the CGMIN module is not ready to use non integer occupations number  
https://github.com/nwchemgit/nwchem/commit/a1d5ac21745cd912bb5235422d368074782fc15c

**Comment by edoapra** [comment]
Code stops when this input combination is used.
https://github.com/nwchemgit/nwchem/commit/a1d5ac21745cd912bb5235422d368074782fc15c
